In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd

In [ ]:
olympic_data = '/content/drive/MyDrive/Coding Dojo/Raw Data/athleteEventsNoPersonal.csv'
olympic_df = pd.read_csv(olympic_data)
olympic_df.head()

ID   Age  Height  ...          Sport                               Event Medal
0   5  21.0   185.0  ...  Speed Skating    Speed Skating Women's 500 metres   NaN
1   5  21.0   185.0  ...  Speed Skating  Speed Skating Women's 1,000 metres   NaN
2   5  25.0   185.0  ...  Speed Skating    Speed Skating Women's 500 metres   NaN
3   5  25.0   185.0  ...  Speed Skating  Speed Skating Women's 1,000 metres   NaN
4   5  27.0   185.0  ...  Speed Skating    Speed Skating Women's 500 metres   NaN

[5 rows x 12 columns]

In [ ]:
# How many rows?
# How many columns?
olympic_df.shape

(40616, 12)

In [ ]:
# Which columns have missing values?
# Why do you think there are so many null values in the Medals column?
olympic_df.isnull().sum()

ID            0
Age        1473
Height     9001
Team          0
NOC           0
Games         0
Year          0
Season        0
City          0
Sport         0
Event         0
Medal     34699
dtype: int64

There are many null values in the Medals column because not every athelete medalled in every event.

In [ ]:
# How many entries correspond to the city of London?
olympic_df['City'].value_counts()

London                    3370
Athina                    2325
Beijing                   2056
Sydney                    2049
Rio de Janeiro            2037
Atlanta                   2037
Barcelona                 1973
Los Angeles               1864
Seoul                     1852
Munich                    1525
Helsinki                  1305
Roma                      1259
Montreal                  1253
Mexico City               1146
Moskva                    1116
Tokyo                     1087
Paris                     1064
Berlin                     907
Sochi                      787
Amsterdam                  757
Torino                     687
Vancouver                  673
Melbourne                  670
Antwerpen                  645
Salt Lake City             639
Stockholm                  600
Nagano                     586
Albertville                545
Lillehammer                541
Innsbruck                  504
Calgary                    393
Sarajevo                   331
Lake Pla

3370 entries correspond to the city of London.

In [ ]:
# What age is the youngest athlete in our sample data?
olympic_df['Age'].min()

11.0

In [ ]:
# How many athletes who participated in the sport "Tug-of-War" were from
# Team "Sweden" in our sample dataset?
sport_filter = olympic_df['Sport'] == 'Tug-Of-War'
team_filter = olympic_df['Team'] == 'Sweden'
tug_df = olympic_df.loc[sport_filter & team_filter, :]
tug_df.head()

ID   Age  Height  ...       Sport                        Event   Medal
10023  34205  22.0     NaN  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN
12681  43105  23.0     NaN  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze
13015  44087  29.0   182.0  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze
16250  55169  23.0     NaN  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN
19048  64266  21.0     NaN  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN

[5 rows x 12 columns]

In [ ]:
tug_df['ID'].value_counts()

44087    1
34205    1
68647    1
64266    1
55169    1
43105    1
Name: ID, dtype: int64

There are 6 athletes who participated in Tug-of-War representing Sweden in our dataset.

In [ ]:
# How many Gold Medals were awarded to athletes who were over 40 years old?
olympic_df['Medal'].value_counts()

Gold      2009
Bronze    1994
Silver    1914
Name: Medal, dtype: int64

In [ ]:
medal_filter = olympic_df['Medal'] == 'Gold'
age_filter = olympic_df['Age'] > 40
gold_df = olympic_df.loc[medal_filter & age_filter, :]
gold_df.shape

(66, 12)

There are 66 Gold medals awarded to athletes over 40 in our sample dataset.

In [ ]:
# Determine the average height of all rows included.
olympic_df['Height'].mean()

175.3811165585956

The average height of all rows is 175.38 cm.

In [ ]:
# Determine the average height of all rows included, but grouped by Event.
olympic_df.groupby(['Event'])[['Height']].mean()

Height
Event                                                     
Alpine Skiing Men's Combined                    178.547170
Alpine Skiing Men's Downhill                    178.211679
Alpine Skiing Men's Giant Slalom                176.739394
Alpine Skiing Men's Slalom                      176.482143
Alpine Skiing Men's Super G                     179.850575
...                                                    ...
Wrestling Women's Flyweight, Freestyle          158.100000
Wrestling Women's Heavyweight, Freestyle        173.444444
Wrestling Women's Light-Heavyweight, Freestyle  170.000000
Wrestling Women's Lightweight, Freestyle        162.666667
Wrestling Women's Middleweight, Freestyle       164.533333

[726 rows x 1 columns]

In [ ]:
# BONUS: determine the average height of all athletes,
# without including duplicates
unique_athlete = olympic_df.drop_duplicates(subset = 'ID', keep = 'first')
unique_athlete.head()

ID   Age  ...                                     Event   Medal
0    5  21.0  ...          Speed Skating Women's 500 metres     NaN
6    6  31.0  ...  Cross Country Skiing Men's 10 kilometres     NaN
14  12  31.0  ...                   Badminton Men's Singles     NaN
15  30  26.0  ...    Rowing Men's Lightweight Double Sculls  Silver
17  36  21.0  ...       Swimming Men's 100 metres Butterfly     NaN

[5 rows x 12 columns]

In [ ]:
unique_athlete['Height'].mean()

176.31818479042698

In [ ]:
unique_athlete.groupby(['Event'])[['Height']].mean()

Height
Event                                                     
Alpine Skiing Men's Combined                    178.800000
Alpine Skiing Men's Downhill                    178.177215
Alpine Skiing Men's Giant Slalom                176.379310
Alpine Skiing Men's Slalom                      175.583333
Alpine Skiing Men's Super G                     178.933333
...                                                    ...
Wrestling Women's Flyweight, Freestyle          158.000000
Wrestling Women's Heavyweight, Freestyle        172.000000
Wrestling Women's Light-Heavyweight, Freestyle  170.000000
Wrestling Women's Lightweight, Freestyle        161.700000
Wrestling Women's Middleweight, Freestyle       165.100000

[609 rows x 1 columns]